### PR TITLE
[FLINK-18756][table-api] Support IF NOT EXISTS for CREATE TABLE statement

### PR DESCRIPTION
**Pull request by leonardBang**
## What is the purpose of the change

* This pull request support `IF NOT EXISTS` for create table statement.

## Brief change log

 - Support parse `IF NOT EXISTS` in `parserImpls.ftl`
 - Simplify `SqlCreateTable.java` 's constructor

## Verifying this change
- Add parse tests in `FlinkSqlParserImplTest.java`
- Add catalog related test in `TableEnvironmentTest.scala`
 
## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl [hunk 1]
```diff
@@ -735,6 +736,11 @@ SqlCreate SqlCreateTable(Span s, boolean replace, boolean isTemporary) :
 {
     <TABLE>
 
+    [
+        LOOKAHEAD(3)
+        <IF> <NOT> <EXISTS> { ifNotExists = true; }
```
Comment (danny0405) [inline flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl:741]: We can use `IfNotExistsOpt` parse block directly.

File: flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/api/TableEnvironmentTest.scala [hunk 2]
```diff
@@ -135,6 +135,82 @@ class TableEnvironmentTest {
...
+        |)
+      """.stripMargin
+    val tableResult1 = tableEnv.executeSql(createTableStmt)
+    assertEquals(ResultKind.SUCCESS, tableResult1.getResultKind)
+    assertTrue(tableEnv.getCatalog(tableEnv.getCurrentCatalog).get()
+      .tableExists(ObjectPath.fromString(s"${tableEnv.getCurrentDatabase}.tbl1")))
+
+    val tableResult2 = tableEnv.executeSql("ALTER TABLE tbl1 SET ('k1' = 'a', 'k2' = 'b')")
+    assertEquals(ResultKind.SUCCESS, tableResult2.getResultKind)
+    assertEquals(
```
Comment (danny0405) [inline flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/api/TableEnvironmentTest.scala:158]: The `ALTER TABLE tbl1 SET` is not necessary, we can merge the test `testExecuteSqlWithCreateTableIfNotExistsTwice` into this one.

**Comment by leonardBang** [comment]
@danny0405 could you have a more look?

**Comment by wuchong** [comment]
Could you add a test in SQL CLI to make sure it also works in SQL CLI? Maybe we can add it in `LocalExecutorITCase`.

**Pull request merged**